pkg/api/types.go

```diff
@@ -466,6 +470,30 @@ type GlusterfsVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// RBDVolumeSource represents a Rados Block Device Mount that lasts the lifetime of a pod
+type RBDVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that represents a collection of Ceph monitors
+	EndpointsName string `json:"endpoints"`
+	// Required: RBDImage is the rados image name
+	RBDImage string `json:"image"`
+	// Required: Filesystem type to mount.
+	// Must be a filesystem type supported by the host operating system.
+	// Ex. "ext4", "xfs", "ntfs"
+	// TODO: how do we prevent errors in the filesystem from compromising the machine
+	FSType string `json:"fsType,omitempty"`
+	// Optional: RBDPool is the rados pool name,default is rbd
+	RBDPool string `json:"pool"`
+	// Optional: RadosUser is the rados user name, default is admin
+	RadosUser string `json:"user"`
+	// Optional: Keyring is the path to key ring for RBDUser, default is /etc/ceph/keyring
+	Keyring string `json:"keyring"`
+	// Optional: Secret is the authentication secret for RBDUser, default is empty.
+	Secret string `json:"secret" description:"secret is authentication secret. If provided, it overrides keyring"`
```

Comment:
> It's possible, but vastly complicates the security story.

I think it makes the security story cleaner, so maybe I did not explain or maybe I am making faulty assumptions.

Assumption: the secret key needed to access a Ceph device is a shared key (or per-device), not a per-user key.  Given that, it makes sense for a PV.PVS.RBD.Secret to be an ObjectRef to a key in an admin-controlled namespace that the user Pod would never see.  a Pod.V.VS.RBD.Secret would be an ObjectRef that must be same-namespace-as-pod.  The main difference being admin controlled vs user controlled.

Now, if my assumption is wrong, and it may well be, then the option to use authenticated storage as a PV is in doubt, unless we define an abstraction around auth that is backend-neutral (unlikely?).